controllers/argocd/notifications.go

```diff
@@ -432,6 +454,62 @@ func (r *ReconcileArgoCD) reconcileNotificationsDeployment(cr *argoproj.ArgoCD,
 
 }
 
+// reconcileNotificationsService will ensure that the Service for the Notifications controller metrics is present.
+func (r *ReconcileArgoCD) reconcileNotificationsService(cr *argoproj.ArgoCD) error {
+
+	var component = "notifications-controller"
+	svc := newServiceWithSuffix(component, component, cr)
+	if argoutil.IsObjectFound(r.Client, cr.Namespace, svc.Name, svc) {
+		// Service found, do nothing
+		return nil
+	}
+
+	svc.Spec.Selector = map[string]string{
+		common.ArgoCDKeyName: nameWithSuffix(component, cr),
+	}
+
+	svc.Spec.Ports = []corev1.ServicePort{
+		{
+			Name:       "notification",
+			Port:       9001,
+			Protocol:   corev1.ProtocolTCP,
+			TargetPort: intstr.FromInt(9001),
+		},
+	}
+
+	if err := controllerutil.SetControllerReference(cr, svc, r.Scheme); err != nil {
+		return err
+	}
+	return r.Client.Create(context.TODO(), svc)
+}
+
+// reconcileNotificationsServiceMonitor will ensure that the ServiceMonitor for the Notifications controller metrics is present.
+func (r *ReconcileArgoCD) reconcileNotificationsServiceMonitor(cr *argoproj.ArgoCD) error {
+
+	name := fmt.Sprintf("%s-%s", cr.Name, "notifications-controller")
+	serviceMonitor := newServiceMonitorWithName(name, cr)
+	if argoutil.IsObjectFound(r.Client, cr.Namespace, serviceMonitor.Name, serviceMonitor) {
```

Comment:
we should add a check for whether prometheus API is available before trying to retrieve a serviceMonitor from the cluster

Reply:
but what if the Prometheus setup is done later?

Reply:
then we can make it so that the API checks are run at the beginning of each reconciliation cycle if needed